src/ripple/app/tx/impl/CashCheck.cpp

```diff
@@ -299,13 +299,13 @@ CashCheck::doApply ()
             // incremental reserve amount in the transfer.  Hence the -1
             // argument.
             STAmount const srcLiquid {xrpLiquid (psb, srcId, -1, viewJ)};
+            auto const optDeliverMin = ctx_.tx[~sfDeliverMin];
 
             // Now, how much do they need in order to be successful?
             STAmount const xrpDeliver {
-                [&sendMax, &srcLiquid] (STTx const& tx)
+                [&sendMax, &srcLiquid, &optDeliverMin] (STTx const& tx)
```

Comment:
Three nits here:
1) I think it formats better with assignment instead of `{}`.
2) This is an immediately invoked lambda, why are we capturing some variable and passing others as parameters. I'd capture them all.
3) Because this is an immediately invoked lambda, there is no danger of dangling references. The explicit lambda capture doesn't provide any value. A default capture would improve the code. (I already know you disagree :-)

Here's my suggestion:
```
            STAmount const xrpDeliver = [&]() {
                // If the check cash specified Amount deliver exactly that.
                if (!optDeliverMin)
                    return ctx_.tx.getFieldAmount(sfAmount);

                return (std::max(*optDeliverMin, std::min(sendMax, srcLiquid)));
            }();
```

Reply:
I've attempted to resolve a couple of these nits with a `[FOLD]` commit.  But I'll add a couple of comments:

1. My inclination is to brace initialize whenever it's reasonable.  When general brace initialization was introduced in C++11, the goal was that everyone could use it everywhere.  And it would be safer that other forms of initialization since it doesn't narrow.  That failed.  Anything with an `initializer_list` constructor (like `vector`) causes problems.  `auto` causes problems (although we hope that improves with C++17).  But in my effort to think as little as possible while coding I reach for `{}` initialization by default unless I know of a reason not to.  Effective Modern C++ Item 7 is my basic guide here.

2. In this particular lambda, the passed parameter is a member of the class, not a local.  I can't capture that.  I can capture `this`, but that passes more information to the lambda that the lambda requires.  So I chose to solve the problem by passing just the member variable that I needed as a parameter.

Hope that helps.

The `[FOLD]` commit resolves nits 2 and 3 by replacing the lambda with a ternary operator (as @HowardHinnant mentioned might be a choice).